src/ape_geth/provider.py

```diff
@@ -471,4 +471,23 @@ def build_command(self) -> List[str]:
 
 # NOTE: The default behavior of EthereumNodeBehavior assumes geth.
 class Geth(EthereumNodeProvider):
-    pass
+    @property
+    def uri(self) -> str:
+        uri = super().uri
+        ecosystem = self.network.ecosystem.name
+        network = self.network.name
+
+        # If we didn't find one in config, look for a public RPC.
+        if not uri or uri == DEFAULT_SETTINGS["uri"]:
+            # Do not override explicit configuration
+            if hasattr(self.config, ecosystem):
```

Comment:
you can also use `.get` on configs

Reply:
Would that improve this somehow?  Same amount of code but it would be a truthy value check instead of a strict bool check, I think.  Unless I'm missing something.

Reply:
I would do this:

```python
if ecosystem in self.config:
```

if it lets you, it reads better.

Reply:
but no, just a nit i guess.
`hasattr` and `getattr` are cool but always feel hacky to me

Reply:
or this:

```python
if cfg := self.config.get(ecosystem)
```

Reply:
[7d3cc1f](https://github.com/ApeWorX/ape/pull/1866/commits/7d3cc1f66005738fbe7fdb3a2ff7ea5c4215a983)